File: Data Structures & Algorithms/min-cost-climbing-stairs/submission-2.java
class Solution {
    public int minCostClimbingStairs(int[] cost) {
        int dp[]=new int[cost.length];
        if(cost.length==1)
            return cost[0];
        if(cost.length==2)
            return Math.min(cost[0],cost[1]);
        dp[0]=cost[0];
        dp[1]=cost[1];
        for(int i=2;i<cost.length;i++){
            dp[i]=Math.min(dp[i-1]+cost[i],dp[i-2]+cost[i]);
        }
        return Math.min(dp[cost.length-2],dp[cost.length-1]);
    }
}

File: Data Structures & Algorithms/eating-bananas/submission-2.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        left = 1
        right = max(piles)
        result = right
        while left <= right:
            k=(left+right)//2
            count=0
            for bananas in piles:
                count+= math.ceil(bananas/k)
            if count<=h:
                result=k
                right=k-1
            else:
                left=k+1
        return result

        